PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/SingleOperationInLoopCheck.java

```diff
@@ -0,0 +1,288 @@
+ */
+public class SingleOperationInLoopCheck extends LocalInspectionTool {
+
+    /**
+     * Build the visitor for the inspection. This visitor will be used to traverse the PSI tree.
+     * @param holder The holder for the problems found
+     * @param isOnTheFly Whether the inspection is running on the fly. If true, the inspection is running as you type.
+     * @return The visitor for the inspection
+     */
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new SingleOperationInLoopVisitor(holder, isOnTheFly);
+    }
+
+    /**
+     * Visitor class to traverse the PSI tree and check for single Azure client operation inside a loop.
+     * The visitor will check for loops of type for, foreach, while, and do-while.
+     * The visitor will check for a Text Analytics client operation inside the loop.
+     * If a Text Analytics client operation is found inside the loop, and the API has a batch alternative, a problem will be registered.
+     */
+    public static class SingleOperationInLoopVisitor extends JavaElementVisitor {
+
+        // Define the holder for the problems found and whether the inspection is running on the fly
+        private static ProblemsHolder holder = null;
+        private final boolean isOnTheFly;
+
+        // Define the logger for the visitor
+        private static final Logger LOGGER = Logger.getLogger(SingleOperationInLoopCheck.class.getName());
+
+        // Define the suggestion message for the problem
+        private static String SUGGESTION = "";
+
+        private static List<String> AVAILABLE_BATCH_METHODS;
+
+        // Load the config file
+        static {
+            try {
+                Map<String, Object> ruleConfigsMap = getRuleConfigs();
+
+                // extract the async return types to check
+                AVAILABLE_BATCH_METHODS = (List<String>) ruleConfigsMap.get("methods_to_check");
+                SUGGESTION = (String) ruleConfigsMap.get("antipattern_message");
+
+            } catch (IOException e) {
+                LOGGER.severe("Failed to load rule configurations");
+            }
+        }
+
+        /**
+         * Constructor for the visitor
+         * @param holder     The holder for the problems found
+         * @param isOnTheFly Whether the inspection is running on the fly. If true, the inspection is running as you type.
+         */
+        public SingleOperationInLoopVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+            this.isOnTheFly = isOnTheFly;
+        }
+
+        /**
+         * Visit the for statement and check for single Azure client operation inside the loop.
+         * @param statement The for statement to check
+         */
+        @Override
+        public void visitForStatement(@NotNull PsiForStatement statement) {
+            checkLoopForTextAnalyticsClientOperation(statement);
```

Comment:
similar comment as above, is this rule only limited to TextAnalytics for now or should we keep it more general for extension.

Reply:
i think specific for now -- it can be an issue for once I finish with everything else